app/objects/secondclass/c_fact.py

```diff
@@ -56,10 +89,65 @@ def __eq__(self, other):
             return self.unique == other.unique
         return False
 
-    def __init__(self, trait, value=None, score=1, collected_by=None, technique_id=None):
+    def __init__(self, name, value=None, source_type=SourceTypes["LEARNED"], source=NOT_SPECIFIED,
+                 restrictions=NOT_SPECIFIED, score=1, technique_id=None,
+                 collected_by=None):  # collected_by will be removed in the near future
         super().__init__()
-        self.trait = trait
+        self.name = name
         self.value = value
+        self.timestamp = datetime.now()
+        self.__source_type = source_type
+        self.source = source
+        self.links = list()
+        self.__restrictions = restrictions
+        self.relationships = list()
         self.score = score
-        self.collected_by = collected_by
         self.technique_id = technique_id
+
+        self.__collected_by = collected_by  # Collected_by will be removed in the near future, succeeded by source/links
+
+    @property
+    def collected_by(self):
+        print(f"collected_by will be deprecated. Please use source/links instead.")
```

Comment:
Do we want to upgrade these deprecation messages to full warnings?

Reply:
If we're going to go live with them, that might be a good idea. At the moment, they're mainly so that I can tell when calls are being made and track down the origins.